source/solvers/postprocessing_cfd.cc

```diff
@@ -35,19 +35,30 @@ double
 calculate_CFL(const DoFHandler<dim> &dof_handler,
               const VectorType &     evaluation_point,
               const double           time_step,
-              const MPI_Comm &       mpi_communicator)
+              const MPI_Comm &       mpi_communicator,
+              const bool             simplex_enabled)
 {
-  const FiniteElement<dim> &fe = dof_handler.get_fe();
-  QGauss<dim>               quadrature_formula(1);
-  const MappingQ<dim>       mapping(fe.degree, false);
-  FEValues<dim>             fe_values(mapping,
+  const FiniteElement<dim> &       fe = dof_handler.get_fe();
+  std::shared_ptr<Quadrature<dim>> quadrature_formula;
+  std::shared_ptr<Mapping<dim>>    mapping;
+  if (simplex_enabled)
+    {
+      mapping = std::make_shared<MappingFE<dim>>(FE_SimplexP<dim>(fe.degree));
+      quadrature_formula = std::make_shared<QGaussSimplex<dim>>(1);
+    }
+  else
+    {
+      mapping            = std::make_shared<MappingQ<dim>>(fe.degree, false);
+      quadrature_formula = std::make_shared<QGauss<dim>>(1);
+    }
+  FEValues<dim> fe_values(*mapping,
```

Comment:
It would be better to pass the  mapping, fe and quadrature as argument instead of re-instantiating them like this. More arguments to functions is not a problem.  This would allow you to remove the simplex argument also.